mythril_core/src/time.rs

```diff
@@ -0,0 +1,165 @@
+use crate::error::Result;
+use crate::tsc;
+
+use core::ops::{Add, Sub};
+use core::time::Duration;
+
+static mut TIME_SRC: Option<&'static mut dyn TimeSource> = None;
```

Comment:
Hmm, apparently `lazy_static` now has `spin_no_std` which allows it to be used no_std. So that might be possible. I'll have to think about it

Reply:
Exploring it a bit. This is possible, but I don't think I like the idea. It leads to strange situations. For example, if the logging subsystem tries to generate a timestamp but the timers haven't been calibrated, it can cause the timers to go _start calibrating_, but this can lead to a lot of problems, the most obvious of which is 'what if the logging is coming from the calibration code'. Ultimately, I think I want the initialization of statics to have a more obviously defined order (and I don't think `lazy_static::initialize` really helps us here).

Reply:
:+1: sounds good